modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp

```diff
@@ -947,10 +947,32 @@ class CoreAudioIODevice   : public AudioIODevice,
             jassert (inputDeviceId != 0);
             device = new CoreAudioInternal (*this, inputDeviceId, true, outputDeviceId != 0);
         }
-        else
+        else if (inputDeviceId == 0)
         {
             device = new CoreAudioInternal (*this, outputDeviceId, false, true);
         }
+        else
+        {
+            NSDictionary* description = @{
+                @(kAudioAggregateDeviceUIDKey) : @"com.juce.aggregate",
+                @(kAudioAggregateDeviceIsPrivateKey) : @(1),
+                @(kAudioAggregateDeviceSubDeviceListKey): @[
+                    @{
+                        @(kAudioSubDeviceUIDKey) : getDeviceUID (inputDeviceId),
+                        @(kAudioSubDeviceDriftCompensationKey) : @(1),
+                        @(kAudioSubDeviceDriftCompensationQualityKey) : @(kAudioSubDeviceDriftCompensationMaxQuality),
+                    },
+                    @{
+                        @(kAudioSubDeviceUIDKey) : getDeviceUID (outputDeviceId),
+                        @(kAudioSubDeviceDriftCompensationKey) : @(1),
+                        @(kAudioSubDeviceDriftCompensationQualityKey) : @(kAudioSubDeviceDriftCompensationMaxQuality),
+                    },
+                ],
+            };
+            OSStatus status = AudioHardwareCreateAggregateDevice ((CFDictionaryRef)description, &aggregateDeviceID);

```

Comment:
Perhaps this merits a comment explaining that we wouldn't get here unless `createDevice` verified that this API is available.